pkg/kubelet/rkt/rkt.go

```diff
@@ -920,10 +927,14 @@ func (r *Runtime) generateRunCommand(pod *api.Pod, uuid, netnsName string) (stri
 		runPrepared = append(runPrepared, "--no-overlay=true")
 	}
 
-	// Network namespace set up in kubelet; rkt networking not used
-	runPrepared = append(runPrepared, "--net=host")
+	// Apply '--net=host' to pod that is running on host network or inside a network namespace.
+	if kubecontainer.IsHostNetworkPod(pod) || r.networkPlugin.Name() != network.DefaultPluginName {
```

Comment:
@feiskyer So when `kubenet` or `cni` network plugin is used, we put `rkt run` in the netns created here. 
And the `rkt run` command itself does `rkt run --net=host`.